src/clj/kixi/hecuba/api/devices.clj

```diff
@@ -92,45 +97,63 @@
       false)))
 
 (defn- ext-type [sensor type-ext]
+  ;;(println "H> kixi.hecuba.api.devices/ext-type")
   (-> sensor
       (update-in [:type] #(str % "_" type-ext))
       (assoc :period "PULSE" :synthetic true :sensor_id (uuid-str))))
 
-(defmulti calculated-sensor (fn [sensor] (when-let [unit (:unit sensor)]
-                                           (.toUpperCase unit))))
+(defmulti calculated-sensor
+  (fn [sensor] (when-let [unit (:unit sensor)]
+                 (.toUpperCase unit))))
 
 (defmethod calculated-sensor "KWH" [sensor]
+  ;;(println "H> kixi.hecuba.api.devices/calculated-sensor KWH")
   (-> sensor
       (assoc :unit "co2" :synthetic true :sensor_id (uuid-str))
       (update-in [:type] #(sensors/output-type-for % "KWH2CO2"))))
 
 (defmethod calculated-sensor "M^3" [sensor]
+  ;;(println "H> kixi.hecuba.api.devices/calculated-sensor M^3")
   (let [kwh-sensor (-> sensor
                        (assoc :unit "kWh" :synthetic true :sensor_id (uuid-str))
                        (update-in [:type] #(sensors/output-type-for % "VOL2KWH")))]
     [kwh-sensor (calculated-sensor kwh-sensor)]))
 
 (defmethod calculated-sensor "FT^3" [sensor]
+  ;;(println "H> kixi.hecuba.api.devices/calculated-sensor M^3")
   (let [kwh-sensor (-> sensor
                        (assoc :unit "kWh" :synthetic true :sensor_id (uuid-str))
                        (update-in [:type] #(sensors/output-type-for % "VOL2KWH")))]
     [kwh-sensor (calculated-sensor kwh-sensor)]))
 
-(defmethod calculated-sensor :default [sensor])
+(defmethod calculated-sensor :default [sensor]
+  ;;(println "H> kixi.hecuba.api.devices/calculated-sensor default")
+  )
 
 (defn create-default-sensors
   "Creates default sensors whenever new device is added: *_differenceSeries for CUMULATIVE,
    and *_co2 for kwh PULSE, etc."
   [body]
+  (println "H> kixi.hecuba.api.devices/create-default-sensors")
   (let [sensors        (:readings body)
-        new-sensors    (map #(case (:period %)
-                               "CUMULATIVE" (ext-type % "differenceSeries")
-                               "PULSE"      (calculated-sensor %)
-                               "INSTANT"    nil
-                               nil) sensors)]
+        new-sensors (case (count sensors)
```

Comment:
are you running your tests against this changed code? The logic doesn't look right to me.